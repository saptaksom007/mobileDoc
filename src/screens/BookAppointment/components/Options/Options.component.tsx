import React, { PureComponent } from 'react'
import { View, StyleSheet, StyleProp, TextStyle } from 'react-native'
import { ButtonGroup } from 'react-native-elements'
import { Heading3 } from 'components/CustomText/CustomText.component'
import { Color } from 'constants/Color'
import CustomTextStyles, {
  fontFamily,
} from 'components/CustomText/CustomText.styles'

export const buttonTextStyle = [
  CustomTextStyles.font,
  CustomTextStyles.small,
  {
    textAlign: 'center',
    paddingHorizontal: 2,
    color: Color.lightText,
  },
]

const styles = StyleSheet.create({
  subtitle: {
    marginLeft: 10,
    color: Color.black,
    fontFamily,
  },
  selectedButtonStyle: {
    backgroundColor: '#8950b7',
  },
  selectedText: {
    color: '#fff',
  },
  containerButtonStyle: {
    // borderRadius: 0,
    height: 45,
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
    shadowColor: '#000',
    shadowOffset: { width: 1, height: 1 },
    shadowOpacity: 0.4,
    shadowRadius: 3,
    elevation: 5,
  },
})

interface Props {
  title: string
  values: string[]
  selectedIndex: number
  onPress(index: number): void
  style?: any
}

const OptButton = ButtonGroup as any

class Options extends PureComponent<Props> {
  static defaultProps = {
    selectedIndex: 0,
    onPress: () => console.warn('Attach a method here.'),
  }
  render() {
    const { title, values, selectedIndex, onPress, style } = this.props
    return (
      <View style={style}>
        {title && <Heading3 style={styles.subtitle}>{title}</Heading3>}
        <OptButton
          selectedButtonStyle={styles.selectedButtonStyle}
          selectedIndex={selectedIndex}
          buttons={values}
          textStyle={buttonTextStyle as StyleProp<TextStyle>}
          containerStyle={styles.containerButtonStyle}
          selectedTextStyle={styles.selectedText}
          onPress={onPress}
        />
      </View>
    )
  }
}

export default Options
