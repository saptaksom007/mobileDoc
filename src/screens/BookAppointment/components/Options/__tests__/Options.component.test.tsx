import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Options from '../Options.component'

it('<Options/> render correctly', () => {
  const tree = renderer.create(<Options title="test" values={[]} />).toJSON()
  expect(tree).toMatchSnapshot()
})
