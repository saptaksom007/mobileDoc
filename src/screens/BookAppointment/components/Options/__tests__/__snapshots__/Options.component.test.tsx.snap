// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options/> render correctly 1`] = `
<View>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontFamily": "TitilliumWeb-SemiBold",
          "fontSize": 18.125,
          "lineHeight": 22,
        },
        Object {
          "color": "#363c40",
          "fontFamily": "TitilliumWeb-Regular",
          "marginLeft": 10,
        },
      ]
    }
  >
    test
  </Text>
  none
</View>
`;
